src/jobserver.h

```diff
@@ -0,0 +1,107 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#define AUTH_KEY  "--jobserver-auth="
+#define FDS_KEY   "--jobserver-fds="
+#define FIFO_KEY  "fifo:"
```

Comment:
Move these to the .cc file, they are not used anywhere else.

Reply:
they would potentially be used in the future `jobserver-win32.cc` for windows

Reply:
If you really want to expose these, convert these to static constexpr members instead for proper scoping, e.g.:

```
// in .h file
struct JobServer {
   static constexpr char kAuthKey[] = "--jobserver-auth=";
   static constexpr char kFdsKey[] = "--jobserver-fds=";
   static constexpr char kFifoKey[] = "fifo:";
};

// in .cc file, to appease picky compilers.
constepxr char JobServer::kAuthKey[];
constexpr char JobServer::kFdsKey[];
constexpr char JobServer::kFifoKey[];
```

This will avoid polluting the preprocessor state of any source or header file that includes this transitively.